features/FEATURE_BLE/targets/TARGET_CORDIO/TESTS/cordio_hci/transport/main.cpp

```diff
@@ -0,0 +1,190 @@
+
+static void hci_driver_rx_dummy_handler(uint8_t* data, uint8_t len) { }
+
+static void hci_driver_rx_reset_handler(uint8_t* data, uint8_t len) {
+    for (size_t i = 0; i < len; ++i) {
+        switch (state) {
+            case WAITING_EVENT_PACKET:
+                if (data[i] == HCI_EVT_TYPE) {
+                    state = WAITING_EVENT_CODE_COMPLETE;
+                } else {
+                    test_result = TEST_RESULT_FAILURE;
+                }
+                break;
+
+            case WAITING_EVENT_CODE_COMPLETE:
+                if (data[i] == HCI_CMD_CMPL_EVT) {
+                    state = WAITING_PARAMETER_LENGTH;
+                } else {
+                    test_result = TEST_RESULT_FAILURE;
+                }
+                break;
+
+            case WAITING_PARAMETER_LENGTH:
+                if (data[i] == RESET_PARAMETER_LENGTH) {
+                    state = WAITING_NUM_HCI_EVT_PACKET;
+                } else {
+                    test_result = TEST_RESULT_FAILURE;
+                }
+                break;
+
+            case WAITING_NUM_HCI_EVT_PACKET:
+                // controler dependent; can be any value, pass on to the next token
+                state = WAITING_OPCODE_LSB;
+                break;
+
+            case WAITING_OPCODE_LSB:
+                if (data[i] == HCI_OPCODE_RESET_LSB) {
+                    state = WAITING_OPCODE_MSB;
+                } else {
+                    test_result = TEST_RESULT_FAILURE;
+                }
+                break;
+
+            case WAITING_OPCODE_MSB:
+                if (data[i] == HCI_OPCODE_RESET_MSB) {
+                    state = WAITING_STATUS;
+                } else {
+                    test_result = TEST_RESULT_FAILURE;
+                }
+                break;
+
+            case WAITING_STATUS:
+                if (data[i] == RESET_EXPECTED_STATUS) {
+                    test_result = TEST_RESULT_SUCCESS;
+                    state = DONE;
+                } else {
+                    test_result = TEST_RESULT_FAILURE;
+                }
+                break;
+        }
+
+        if (test_result != TEST_RESULT_TIMEOUT_FAILURE) {
+            CordioHCIHook::set_data_received_handler(hci_driver_rx_dummy_handler);
+            sem.release();
+            return;
+        }
+    }
+}
+
+void test_reset_command() {
+    CordioHCIDriver& driver = CordioHCIHook::get_driver();
+    CordioHCITransportDriver& transport_driver = CordioHCIHook::get_transport_driver();
+
+    driver.initialize();
+
+    CordioHCIHook::set_data_received_handler(hci_driver_rx_reset_handler);
+    transport_driver.write(HCI_CMD_TYPE, sizeof(reset_cmd), reset_cmd);
+    sem.wait(RESET_COMMAND_TIMEOUT);
+    CordioHCIHook::set_data_received_handler(hci_driver_rx_dummy_handler);
+
+    driver.terminate();
+
+    TEST_ASSERT_EQUAL(TEST_RESULT_SUCCESS, test_result);
+    TEST_ASSERT_EQUAL(DONE, state);
+}
+
+Case cases[] = {
+    Case("Test reset command", test_reset_command),
```

Comment:
Could you add a second test that issues multiple reset commands + `initialize()`/`terminate()`?